ligo/src/treasuryDAO.mligo

```diff
@@ -0,0 +1,116 @@
+// SPDX-FileCopyrightText: 2020 TQ Tezos
+// SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+#include "defaults.mligo"
+#include "types.mligo"
+#include "proposal.mligo"
+#include "helper/unpack.mligo"
+#include "base_DAO.mligo"
+
+#include "treasuryDAO/types.mligo"
+
+// -------------------------------------
+// Helpers
+// -------------------------------------
+
+let unpack_transfer_type_list (b: bytes) : transfer_type list =
+  match ((Bytes.unpack b) : ((transfer_type list) option)) with
+    Some (v) -> v
+  | None -> (failwith "UNPACKING_NOT_NAT_OPTION_TYPE" : transfer_type list)
+
+// -------------------------------------
+// Configuration Lambdas
+// -------------------------------------
+
+let treasury_DAO_proposal_check (params, store : propose_params * storage) : bool =
+  let proposal_size = Bytes.size(Bytes.pack(params.proposal_metadata)) in
+  let frozen_scale_value = unpack_nat(unpack_key("a", store.extra)) in
+  let frozen_extra_value = unpack_nat(unpack_key("b", store.extra)) in
+  let max_proposal_size = unpack_nat(unpack_key("s_max", store.extra)) in
+  let requred_token_lock = frozen_scale_value * proposal_size + frozen_extra_value in
+
+  if (params.frozen_token = requred_token_lock) && (proposal_size < max_proposal_size) then
+    let ts = unpack_transfer_type_list(unpack_key("transfers", params.proposal_metadata)) in
+    let is_all_transfers_valid (is_valid, transfer_type: bool * transfer_type) =
+      match transfer_type with
+          Token_transfer_type tt -> is_valid
+        | Xtz_transfer_type xt ->
+            let min_xtz_amount = unpack_tez(unpack_key("x", store.extra)) in
+            let max_xtz_amount = unpack_tez(unpack_key("z", store.extra)) in
```

Comment:
Isn't it better to put these two lookup/unpack calls outside of the `is_all_transfers_valid`, since it apepar to remain constant and also appear to be a bit costly.